numpy/_core/src/umath/string_ufuncs.cpp

```diff
@@ -363,10 +369,24 @@ string_replace(Buffer<enc> buf1, Buffer<enc> buf2, Buffer<enc> buf3, npy_int64 c
             buf1 += len2;
         }
     }
+    else {  // If match string empty, interleave.
+        while (count > 0) {
+            buf3.buffer_memcpy(out, len3);
+            out += len3;
+
+            if (--count <= 0) {
+                break;
+            }
+
+            buf1.buffer_memcpy(out, 1);
```

Comment:
OK, I think I finally got what you meant - if `buf1` itself is empty, then this copy is a problem.